packages/ckeditor5-engine/src/conversion/mapper.ts

```diff
@@ -893,16 +894,20 @@ export class MapperCache extends /* #__PURE__ */ EmitterMixin() {
 			// `Mapper#getModelLength()` are implemented so that parents are cached before their children.
 			//
 			// So, don't create new cache if one already exists. Instead, only save `_nodeToCacheListIndex` value for the related view node.
-			if ( viewOffset > 0 ) {
```

Comment:
When it comes to removing this `if`, I figured out that the way `save()` is used, we should never pass `viewOffset == 0` here. Running all unit tests confirmed this. There were specific unit tests for mapper caching which passed `viewOffset = 0` here, but I've changed them as it was an incorrect use of the method.